### PR TITLE
feat: Added sql managed instances

### PR DESCRIPTION
**Pull request by amanenk**
Added sql managed instances and its relations required to fulfill HIPAA queries

### CODE REVIEW

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, requires fix to the migrations (v0.5.0) + e2e testing, they should work now on PR once you rebase from main.

**Review by spangenberg — APPROVED** [review]
LGTM, apart from the migration file name

File: resources/services/sql/managed_databases.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func SqlManagedDatabases() *schema.Table {
+	return &schema.Table{
+		Name:         "azure_sql_managed_databases",
+		Description:  "ManagedDatabase a managed database resource.",
+		Resolver:     fetchSqlManagedDatabases,
+		Multiplex:    client.SubscriptionMultiplex,
+		DeleteFilter: client.DeleteSubscriptionFilter,
```
Comment (irmatov) [inline resources/services/sql/managed_databases.go:18]: As this is a relation on instances there is no need for Multiplex nor DeleteFilter.

File: resources/services/sql/managed_databases.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+			},
+			{
+				Name:        "restorable_dropped_database_id",
+				Description: "The restorable dropped database resource id to restore when creating this database.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ManagedDatabaseProperties.RestorableDroppedDatabaseID"),
+			},
+			{
+				Name:        "storage_container_sas_token",
+				Description: "Conditional",
```
Comment (irmatov) [inline resources/services/sql/managed_databases.go:91]: nit: incomplete description?

File: resources/services/sql/managed_instances.go [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+			},
+			{
+				Name:        "identity_tenant_id",
+				Description: "The Azure Active Directory tenant id.",
+				Type:        schema.TypeUUID,
+				Resolver:    schema.PathResolver("Identity.TenantID"),
+			},
+			{
+				Name:        "sku_name",
+				Description: "The name of the SKU, typically, a letter + Number code, e.g",
```
Comment (irmatov) [inline resources/services/sql/managed_instances.go:47]: nit: description can be cleaned up a bit (remove trailing `, e.g`)

File: resources/services/sql/managed_instances.go [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Sku.Name"),
+			},
+			{
+				Name:        "sku_tier",
+				Description: "The tier or edition of the particular SKU, e.g",
```
Comment (irmatov) [inline resources/services/sql/managed_instances.go:53]: nit: same as above

File: resources/services/sql/managed_instances.go [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "Possible values include: 'ProvisioningState1Creating', 'ProvisioningState1Deleting', 'ProvisioningState1Updating', 'ProvisioningState1Unknown', 'ProvisioningState1Succeeded', 'ProvisioningState1Failed'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ManagedInstanceProperties.ProvisioningState"),
+			},
+			{
+				Name:        "managed_instance_create_mode",
+				Description: "Specifies the mode of database creation.  Default: Regular instance creation.  Restore: Creates an instance by restoring a set of backups to specific point in time",
```
Comment (irmatov) [inline resources/services/sql/managed_instances.go:83]: nit: description can be shortened.

**Review by irmatov — CHANGES_REQUESTED** [review]
Remove unnecessary Multiplex and DeleteFilter, otherwise LGTM (there are a few other nits around descriptions).

File: docs/tables/azure_sql_managed_instance_private_endpoint_connections.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+ManagedInstancePecProperty a private endpoint connection under a managed instance
+## Columns
+| Name        | Type           | Description  |
+| ------------- | ------------- | -----  |
+|managed_instance_cq_id|uuid|Unique CloudQuery ID of azure_sql_managed_instances table (FK)|
+|id|text|Resource ID.|
+|private_endpoint_id|text|Resource id of the private endpoint.|
+|private_link_service_connection_state_status|text|The private link service connection status.|
+|private_link_service_connection_state_description|text|The private link service connection description.|
+|private_link_service_connection_state_actions_required|text|The private link service connection description.|
```
Comment (roneli) [inline docs/tables/azure_sql_managed_instance_private_endpoint_connections.md:12]: connection_status, connection_description, connection_actions_required

File: resources/services/sql/managed_databases.go [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v4.0/sql"
+	"github.com/cloudquery/cq-provider-azure/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func SqlManagedDatabases() *schema.Table {
```
Comment (roneli) [inline resources/services/sql/managed_databases.go:12]: nit: would make this unexported.

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, minor nits and we can merge.

File: docs/tables/azure_sql_managed_instances.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+|sku_name|text|The name of the SKU, typically, a letter + Number code, eg|
+|sku_tier|text|The tier or edition of the particular SKU, eg|
+|sku_size|text|Size of the particular SKU|
+|sku_family|text|If the service has different generations of hardware, for the same SKU, then that can be captured here|
+|sku_capacity|integer|Capacity of the particular SKU|
+|provisioning_state|text|Possible values include: 'ProvisioningState1Creating', 'ProvisioningState1Deleting', 'ProvisioningState1Updating', 'ProvisioningState1Unknown', 'ProvisioningState1Succeeded', 'ProvisioningState1Failed'|
+|managed_instance_create_mode|text|Specifies the mode of database creation|
+|fully_qualified_domain_name|text|The fully qualified domain name of the managed instance|
+|administrator_login|text|Administrator username for the managed instance|
+|administrator_login_password|text|The administrator login password (required for managed instance creation)|
```
Comment (roneli) [inline docs/tables/azure_sql_managed_instances.md:20]: would maybe not fetch this field `administrator_login_password` , although I am pretty sure it doesn't get fetched in read.

File: resources/services/sql/managed_instances.go [hunk 9]
```diff
@@ -0,0 +1,488 @@
...
+						Description: "The private link service connection description",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("Properties.PrivateLinkServiceConnectionState.Description"),
+					},
+					{
+						Name:        "private_link_service_connection_state_actions_required",
+						Description: "The private link service connection description",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("Properties.PrivateLinkServiceConnectionState.ActionsRequired"),
+					},
```
Comment (roneli) [inline resources/services/sql/managed_instances.go:281]: didn't we say in previous review to shorten these?

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, to more minor nits.

**Review by roneli — APPROVED** [review]

**Pull request merged**